homeassistant/components/reolink/config_flow.py

```diff
@@ -96,7 +96,37 @@ async def async_step_reauth_confirm(
     async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
         """Handle discovery via dhcp."""
         mac_address = format_mac(discovery_info.macaddress)
-        await self.async_set_unique_id(mac_address)
+        existing_entry = await self.async_set_unique_id(mac_address)
+        if (
+            existing_entry
+            and CONF_PASSWORD in existing_entry.data
+            and existing_entry.data[CONF_HOST] != discovery_info.ip
+        ):
+            # check if the camera is reachable at the new IP
```

Comment:
We should also abort if the config entry for the device is LOADED and working before trying to probe the new ip

Reply:
Check if its LOADED and working, ie not failing to update.

unifiprotect does exactly this for the same reason in its config flow discovery path

Reply:
@bdraco thanks for pointing me in the direction of unifiprotect!
First I did not understood what you ment.
I now have implemented the additional check.